src/porepy/applications/verification_setups/manu_flow_incomp_frac.py

```diff
@@ -338,85 +410,192 @@ def boundary_values(self, sd_rock: pp.Grid) -> np.ndarray:
+            approx_array=out.approx_rock_pressure,
             is_scalar=True,
             is_cc=True,
         )
 
-        # Rock flux
-        self.exact_rock_flux = ex.rock_flux(sd_rock)
-        self.approx_rock_flux = data_rock[pp.STATE][q_name]
-        self.error_rock_flux = self.relative_l2_error(
-            grid=sd_rock,
-            true_array=self.exact_rock_flux,
-            approx_array=self.approx_rock_flux,
-            is_scalar=True,
-            is_cc=False,
-        )
-
         # Fracture pressure
-        self.exact_frac_pressure = ex.fracture_pressure(sd_frac)
-        self.approx_frac_pressure = data_frac[pp.STATE][p_name]
-        self.error_frac_pressure = self.relative_l2_error(
+        out.exact_frac_pressure = exact_sol.fracture_pressure(sd_frac)
+        out.approx_frac_pressure = data_frac[pp.STATE][p_name]
+        out.error_frac_pressure = self.relative_l2_error(
             grid=sd_frac,
-            true_array=self.exact_frac_pressure,
-            approx_array=self.approx_frac_pressure,
+            true_array=out.exact_frac_pressure,
+            approx_array=out.approx_frac_pressure,
             is_scalar=True,
             is_cc=True,
         )
 
+        # Rock flux
+        out.exact_rock_flux = exact_sol.rock_flux(sd_rock)
+        out.approx_rock_flux = data_rock[pp.STATE][q_name]
+        out.error_rock_flux = self.relative_l2_error(
+            grid=sd_rock,
+            true_array=out.exact_rock_flux,
+            approx_array=out.approx_rock_flux,
+            is_scalar=True,
+            is_cc=False,
+        )
+
         # Fracture flux
-        self.exact_frac_flux = ex.fracture_flux(sd_frac)
-        self.approx_frac_flux = data_frac[pp.STATE][q_name]
-        self.error_frac_flux = self.relative_l2_error(
+        out.exact_frac_flux = exact_sol.fracture_flux(sd_frac)
+        out.approx_frac_flux = data_frac[pp.STATE][q_name]
+        out.error_frac_flux = self.relative_l2_error(
             grid=sd_frac,
-            true_array=self.exact_frac_flux,
-            approx_array=self.approx_frac_flux,
+            true_array=out.exact_frac_flux,
+            approx_array=out.approx_frac_flux,
             is_scalar=True,
             is_cc=False,
         )
 
-        # Mortar flux
-        self.exact_intf_flux = ex.mortar_fluxes(intf)
-        self.approx_intf_flux = data_intf[pp.STATE][lmbda_name]
-        self.error_intf_flux = self.relative_l2_error(
+        # Interface flux
+        out.exact_intf_flux = exact_sol.interface_flux(intf)
+        out.approx_intf_flux = data_intf[pp.STATE][lmbda_name]
+        out.error_intf_flux = self.relative_l2_error(
             grid=intf,
-            true_array=self.exact_intf_flux,
-            approx_array=self.approx_intf_flux,
+            true_array=out.exact_intf_flux,
+            approx_array=out.approx_intf_flux,
             is_scalar=True,
             is_cc=True,
         )
 
+        return out
+
+
+class SetupUtilities:
```

Comment:
Should this be moved? E.g. to verification_utils?

Reply:
I renamed the classes and recycled utilities among verification setups. However, I don't think it is a good idea to use generic plotting methods.

Reply:
I agree.